filters/auth/tokenintrospection.go

```diff
@@ -333,9 +341,12 @@ func (f *tokenintrospectFilter) Request(ctx filters.FilterContext) {
 
 func (f *tokenintrospectFilter) Response(filters.FilterContext) {}
 
-// Close cleans-up the quit channel used for this spec
+// Close cleans-up the quit channel used for this filter
 func (f *tokenintrospectFilter) Close() {
+	f.authClient.mu.Lock()
```

Comment:
(not important, i think it is fair to expect that the Close is called only from one goroutine)

Reply:
better being safe in this case!
I think on teardown we have plenty of time, no problem for production traffic if we make sure we do not panic because of the Close()

Reply:
Accepted.